python/ql/src/semmle/python/ApiGraphs.qll

```diff
@@ -300,11 +303,18 @@ module API {
       MkRoot() or
       /** An abstract representative for imports of the module called `name`. */
       MkModuleImport(string name) {
-        imports(_, name)
+        (
+          imports(_, name)
+          or
+          // When we `import foo.bar.baz` we want to create API graph nodes also for the prefixes
+          // `foo` and `foo.bar`:
+          name = any(ImportExpr e | not e.isRelative()).getAnImportedModuleName()
+        ) and
+        // Ignore the following module name, as we alias `__builtin__` to `builtins` elsewhere
+        name != "__builtin__"
         or
-        // When we `import foo.bar.baz` we want to create API graph nodes also for the prefixes
-        // `foo` and `foo.bar`:
-        name = any(ImportExpr e | not e.isRelative()).getAnImportedModuleName()
+        // The `builtins` module should always be implicitly available
+        name = "builtins"
       } or
```

Comment:
I cannot add a suggestion to deleted lines(?!) It is not of great import, but I wonder if it reads simpler like this:
```
MkModuleImport(string name) {
  // Ignore the following module name, as we alias `__builtin__` to `builtins` elsewhere
  name != "__builtin__" and
  (
     // The `builtins` module should always be implicitly available
     name = "builtins"
     or
     imports(_, name)
     or
     // When we `import foo.bar.baz` we want to create API graph nodes also for the prefixes
     // `foo` and `foo.bar`:
     name = any(ImportExpr e | not e.isRelative()).getAnImportedModuleName()
  )
} or
```

Reply:
:+1: Done.